core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -101,6 +101,7 @@ class LogCleaner(initialConfig: CleanerConfig,
                  time: Time = Time.SYSTEM) extends Logging with BrokerReconfigurable {
   // Visible for test.
   private[log] val metricsGroup = new KafkaMetricsGroup(this.getClass)
+  activateMetrics()
```

Comment:
As it gets called in `startup`, do we need to call it in construction?

Reply:
If we remove the activate function from the construction, some of the test cases would fail. The metric would not be initialized. Is there a better way to refactor that? thanks!

Reply:
> some of the test cases would fail. The metric would not be initialized. Is there a better way to refactor that? thanks!

Could you please share the test cases to me? Maybe they do not call `startup`

Reply:
Thanks for the reminder. Added the `startup()` back to test functions.